### PR TITLE
XRC-9: Composite content type

### PR DESCRIPTION
**Pull request by mkobetic**
Based on XIP-5 this XRC proposes a new content type `xtmp.org/composite:1.0` for multi-part messages where the parts can be of any content type and the composite parts can be arbitrarily nested.

Reference implementation https://github.com/xmtp/xmtp-js/pull/96

### CODE REVIEW

**Review by neekolas — APPROVED** [review]
I really like how simple this is

File: XIPs/xip-composite-content-type.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+```
+
+## Rationale
+
+It is unclear at this point what particular aspects of multipart content will be relevant in practice, consequently this initial proposal is about as simple as it gets. Subsequent revisions can add more features, although there is some expectation that more specialized content types will be preferred for specific use-cases. Protobuf has been chosen to keep the implementation simple, which should support the educational goals of this proposal.
+
+## Reference Implementation
+
+The reference implementation is intended to be part of the XMTP SDK. [https://github.com/xmtp/xmtp-js/pull/96](https://github.com/xmtp/xmtp-js/pull/96). The codec will not be part of the default Client setup, clients wishing to employ this codec SHOULD register this codec explicitly.
```
Comment (saulmc) [inline XIPs/xip-composite-content-type.md:75]: Given the educational goal of this XIP, would it make sense to provide an example of what it means to "register this codec explicitly"? Or is it obvious to the developer what to do?

**Review by jazzz — APPROVED** [review]
Extremely easy to follow. 👍

**Comment by saulmc** [comment]
@mkobetic This should probably also include an update to the README that adds this XIP to the table.

**Review by saulmc — APPROVED** [review]

File: XIPs/xip-composite-content-type.md (hunk 1 above)
Comment (mkobetic) [reply XIPs/xip-composite-content-type.md:75]: Good point, I won't do it here since it applies to any codec, but I will document it in the SDK readme.

**Comment by mkobetic** [comment]
> @mkobetic This should probably also include an update to the README that adds this XIP to the table.

@saulmc is it the author or the editor that should do that? With XIP-5 @mg0716 merged the PR and updated the table.

Speaking of which I need an editor for this XRC 🙏

**Comment by mg0716** [comment]
> is it the author or the editor that should do that?

The editor (me) would be doing that.

**Review by mg0716 — APPROVED** [review]
nice

Looks good @mkobetic 

---

Additionally I updated the front matter to re-include a specific `xip` column, as unfortunately the `:` within the title column was preventing all front matter from rendering properly.

**Pull request merged**